### PR TITLE
feat(injector): add ephemeral-storage resource request/limit for sidecars

### PR DESCRIPTION
**Pull request by jijiechen**
fixes https://github.com/kumahq/kuma/issues/4946

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues 
  - https://github.com/kumahq/kuma/issues/4946
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS 
  - Confirmed
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s)
  - test case updated
  - Don't forget `ci/` labels to run additional/fewer tests
    - No need
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? 
  - No
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) 
  - No

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/runtime/k8s/webhooks/injector/injector.go [hunk 1]
```diff
@@ -73,27 +73,10 @@ type KumaInjector struct {
 }
 
 func (i *KumaInjector) InjectKuma(ctx context.Context, pod *kube_core.Pod) error {
-	ns, err := i.namespaceFor(ctx, pod)
-	if err != nil {
-		return errors.Wrap(err, "could not retrieve namespace for pod")
-	}
 	logger := log.WithValues("pod", pod.GenerateName, "namespace", pod.Namespace)
-	// Log deprecated annotations
-	for _, d := range metadata.PodAnnotationDeprecations {
```
Comment (jijiechen) [inline pkg/plugins/runtime/k8s/webhooks/injector/injector.go:82]: Refactoring: moved to the separate file `precheck.go`

File: pkg/plugins/runtime/k8s/webhooks/injector/injector.go [hunk 2]
```diff
@@ -404,6 +301,8 @@ func (i *KumaInjector) NewSidecarContainer(
 	}
 
 	container.Name = k8s_util.KumaSidecarContainerName
+	container.Resources.Requests[kube_core.ResourceEphemeralStorage] = pointer.Deref(kube_api.NewScaledQuantity(10, kube_api.Mega))
+	container.Resources.Limits[kube_core.ResourceEphemeralStorage] = pointer.Deref(kube_api.NewScaledQuantity(10, kube_api.Mega))
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/runtime/k8s/webhooks/injector/injector.go:305]: should this be added only for injected sidecar or also for builtin gateway? CC @michaelbeaumont If for both, it's better to place it in `NewContainer()` func

Comment (jijiechen) [reply pkg/plugins/runtime/k8s/webhooks/injector/injector.go]: I've moved this into `factory.go` so it can be shared for both sidecars and gateways

**Review by jakubdyszkiewicz — APPROVED** [review]

File: pkg/plugins/runtime/k8s/containers/factory.go [hunk 3]
```diff
@@ -158,12 +159,14 @@ func (i *DataplaneProxyFactory) NewContainer(
...
+				kube_core.ResourceCPU:              kube_api.MustParse(i.ContainerConfig.Resources.Requests.CPU),
+				kube_core.ResourceMemory:           kube_api.MustParse(i.ContainerConfig.Resources.Requests.Memory),
+				kube_core.ResourceEphemeralStorage: pointer.Deref(kube_api.NewScaledQuantity(50, kube_api.Mega)),
 			},
 			Limits: kube_core.ResourceList{
-				kube_core.ResourceCPU:    kube_api.MustParse(i.ContainerConfig.Resources.Limits.CPU),
-				kube_core.ResourceMemory: kube_api.MustParse(i.ContainerConfig.Resources.Limits.Memory),
+				kube_core.ResourceCPU:              kube_api.MustParse(i.ContainerConfig.Resources.Limits.CPU),
+				kube_core.ResourceMemory:           kube_api.MustParse(i.ContainerConfig.Resources.Limits.Memory),
+				kube_core.ResourceEphemeralStorage: pointer.Deref(kube_api.NewScaledQuantity(1, kube_api.Giga)),
```
Comment (michaelbeaumont) [inline pkg/plugins/runtime/k8s/containers/factory.go:169]: What's the advantage and implications of setting a limit?

Comment (jijiechen) [reply pkg/plugins/runtime/k8s/containers/factory.go:169]: Technically, setting a limit to every resource allocation is a "good manner" to make sure we don't over consume. The implication is when we exceed that limit, the pod get evicted. 

Right now, the [default allowed logging](https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration) on Kubernetes clusters is, 5 x 10Mi

Comment (michaelbeaumont) [reply pkg/plugins/runtime/k8s/containers/factory.go:169]: Actually [we shouldn't even be setting CPU limits](https://home.robusta.dev/blog/stop-using-cpu-limits). What if users configure access logging to file in `/tmp` for example, now the Pod will be evicted if this file hits 10GiB?

**Pull request merged**